src/cfi.rs

```diff
@@ -629,22 +630,23 @@ impl<'bases, 'input, Endian, Section> FallibleIterator
     }
 }
 
+struct CfiEntryCommon<'input, Endian>
+    where Endian: Endianity
+{
+    length: u64,
+    format: Format,
+    cie_offset_input: EndianBuf<'input, Endian>,
+    cie_id_or_offset: u64,
+    rest: EndianBuf<'input, Endian>,
+}
+
 /// Parse the common start shared between both CIEs and FDEs. If we find the
 /// end-of-entries sentinel, return `Ok(None)`. Otherwise, return
-/// `Ok(Some(tuple))`, where `tuple` is of the form:
-///
-///   (next_entry_input, (length,
-///                       format,
-///                       cie_offset_input,
-///                       cie_id_or_offset,
-///                       rest_of_this_entry_input))
-fn parse_cfi_entry_common<'input, Endian, Section>(input: EndianBuf<'input, Endian>)
-                                                   -> Result<Option<(EndianBuf<'input, Endian>,
-                                                                     (u64,
-                                                                      Format,
-                                                                      EndianBuf<'input, Endian>,
-                                                                      u64,
-                                                                      EndianBuf<'input, Endian>))>>
+/// `Ok(Some(tuple))`, where `tuple.0` is the start of the next entry and
+/// `tuple.1` is the parsed CFI entry data.
+fn parse_cfi_entry_common<'input, Endian, Section>
```

Comment:
Curious that we only call this from one place (at least as far as I can tell from what places had to be modified to use `CfiEntryCommon`). I guess it'll be inlined though, so it's okay.